### DISCUSSION TITLE
Meeting notes 05/06

### DISCUSSION THREAD

**Discussion by Ivanderkelen** [CORDEX-be2/ValEnsPy#54] [Meeting minutes]
## Agenda

1. ERA5 input processor: open questions #62 

- download script Hortense: also in helper scripts within Valenspy (but installation of cds api and manual step necessary) @ivanderkelen 
2. CLIMATE_GRID input generator: include also convertor script from raw R data into Valenspy? 

3. Input path generator: need for standardized paths @kobebryant432 #69 

4. domain vs region: naming? Europe/Belgium. 

5. investigation for paid subscription by @Ivanderkelen #66. Is it worth it?  --> take up later again. 

## Meeting notes
- "domain" vs "region": we decided "region" denotes a loosely defined region (eg ERA5 download description, "europe", "belgium", covering those areas but no stringent X-Y coordinates and grid resolution. Domain are actually defined domains, eg the EURO-CORDEX domains EUR-11, EUR-44 etc. 
- ERA5 input processor: #62. Decided to include download script in ancilliary_scripts: not formally part of Valenspy but given as information to users. The conversion script for CLIMATE GRID will also go into this directory. 

- ❗ the grid in the ERA5 and ERA5-Land download scripts of Bert is a user-defined grid, and does not correspond to the standard ERA5 grid. We will want to change this in the future, to have "as raw data" as possible, but for now no priority. 

 5. investigation for paid subscription by @Ivanderkelen #66. Is it worth it?  --> take up later again, no priority now. 

@kobebryant432 feel free to complement. Tagging @nicghil and @bertvs1 so that they are up to date as well 😄. 

## Actions

- [ ] implement CLIMATE GRID input convertor @Ivanderkelen 
- [x] make path generator more robust @kobebryant432 
- [x] Separate ERA5 and ERA5-Land convertor functions. @Ivanderkelen 
- [ ] List global data attributes necessary to include in CF checker @Ivanderkelen 
- [x] Apply domain vs region change. @Ivanderkelen 

### Priorities

**Comment by bertvs1** [comment]
Small note on the "domain" vs "region". The EUR-11 and EUR-44 from CORDEX have fixed boundaries but do not have fixed gridpoints so results for different models are not directly intercomparable. If "domain" is intended to indicate the grid on which the final comparison/analysis will be done, I think the similarity with EUR-11 and EUR-44 is not correct.

**Reply by Ivanderkelen** [reply]
Good point, Bert. I think it is important to further clarify what we exactly mean with "domain" at some point, but at least "region" is more general and very loose.

**Comment by bertvs1** [comment]
Has there been a discussion on the open question of [issue #62](https://github.com/CORDEX-be2/ValEnsPy/issues/62): _The ERA5 postprocessing: include aggregation to daily data in Valenspy package? eg by translating R scripts Bert above? (What with variable naming?) Self defined, eg prhmax (Daily Maximum of hourly precipitation). Or leave it up to user itself._
I would say this may be a quite important issue for simple variables like tasmin and tasmax.

**Reply by Ivanderkelen** [reply]
Not yet, we also even questioned the file structure and download of ERA5 and ERA5-Land (need to download on the original ERA5 grid). The conclusion was that at some point the whole ERA5 structure might need to be revised (including potential postprocessing like day averaging).
